### PR TITLE
Speculative fix for duplicate stateful views in reuse pool

### PR DESCRIPTION
**Pull request by ocrickard**
@conradev this is my stab at fixing the issue you reported. Mind patching it and testing if it fixes your issue?

### CODE REVIEW

File: ComponentKit/StatefulViews/CKStatefulViewComponentController.mm [hunk 1]
```diff
@@ -131,13 +131,13 @@ - (void)_presentStatefulView
...
 {
-  if ([self canRelinquishStatefulView]) {
+  if ([self canRelinquishStatefulView] && _statefulView) {
     [[CKStatefulViewReusePool sharedPool]
      enqueueStatefulView:_statefulView
      forControllerClass:[self class]
      context:_statefulViewContext
      mayRelinquishBlock:^BOOL{
-       if (!self->_mounted && [self canRelinquishStatefulView]) {
+       if (self->_statefulView && !self->_mounted && [self canRelinquishStatefulView]) {
```
Comment (ocrickard) [inline ComponentKit/StatefulViews/CKStatefulViewComponentController.mm]: Here's the issue: _relinquishStatefulViewIfPossible can be called multiple times. I missed that the previous version of this code was using the `if (_statefulView)` check as a way to guarantee that only one block won and actually enqueued the view into the reuse pool.

So what would happen here is that the stateful view could be enqueued into the reuse pool as many times as this method was called, which could result in a stateful view being used for more than one controller at a time.

File: ComponentKit/StatefulViews/CKStatefulViewComponentController.mm [hunk 2]
```diff
@@ -131,13 +131,13 @@ - (void)_presentStatefulView
 
 - (void)_relinquishStatefulViewIfPossible
 {
-  if ([self canRelinquishStatefulView]) {
+  if ([self canRelinquishStatefulView] && _statefulView) {
```
Comment (conradev) [inline ComponentKit/StatefulViews/CKStatefulViewComponentController.mm]: Might want to flip this boolean check

Comment (ocrickard) [inline ComponentKit/StatefulViews/CKStatefulViewComponentController.mm]: Sure, good point.

**Comment by ocrickard** [comment]
Confirmed with @conradev that this fixes the issue that he's seeing. Since it's safe, I'd like to merge this.

**Comment by ocrickard** [comment]
@facebook-github-bot import

**Comment by ocrickard** [comment]
Landed internally, build is green. Merging.

**Pull request merged**